mods/storage/src/storage.ts

```diff
@@ -146,44 +183,43 @@ export default class Storage extends FonosService {
    *   console.log(result)
    * }).catch(e => console.error(e))  // an error occurred
    */
-  async getObjectURL(request: GetObjectURLRequest): Promise<any> {
-    return new Promise((resolve, reject) => {
-      let bucket: StoragePB.GetObjectURLRequest.Bucket;
-      switch (request.bucket) {
-        case "apps":
-          bucket = StoragePB.GetObjectURLRequest.Bucket.APPS;
-          break;
-        case "recordings":
-          bucket = StoragePB.GetObjectURLRequest.Bucket.RECORDINGS;
-          break;
-        case "public":
-          bucket = StoragePB.GetObjectURLRequest.Bucket.PUBLIC;
-          break;
-      }
-
-      const gour = new StoragePB.GetObjectURLRequest();
-      gour.setFilename(request.filename);
-      gour.setBucket(bucket);
-      gour.setAccessKeyId(request.accessKeyId);
-
-      super
-        .getService()
-        .getObjectURL(gour, super.getMeta(), (err: any, res: any) => {
-          if (err) {
-            reject(err as Error);
-          } else {
-            resolve({
-              url: res.getUrl()
-            });
-          }
-        });
-    }).catch((e) => {
-      throw e;
-    });
+  private async getObjectURLInternal(
+    request: GetObjectURLRequest
+  ): Promise<string> {
+    const result = await this.getService()
+      .getObjectURL()
+      .sendMessage(utils.getObjectServiceUtils(request));
+
+    return result.getUrl();
   }
 
   // Internal API
-  uploadObjectSync(request: UploadObjectRequest) {
+  /**
+   * Upload an object to Fonos Object Storage subsystem.
```

Comment:
Consider enhancing comment to clarify that this is a synchronous version of another function.